iothub_service_client/tests/iothub_msging_ll_ut/iothub_messaging_ll_ut.c

```diff
@@ -1512,878 +1253,520 @@ BEGIN_TEST_SUITE(iothub_messaging_ll_ut)
-
-        STRICT_EXPECTED_CALL(singlylinkedlist_item_get_value(IGNORED_PTR_ARG));
-
-        STRICT_EXPECTED_CALL(singlylinkedlist_get_next_item(IGNORED_PTR_ARG));
-
-        STRICT_EXPECTED_CALL(gballoc_free(IGNORED_PTR_ARG));
-
+        for (int i = 0; i < array_count; i++)
+        {
+            STRICT_EXPECTED_CALL(singlylinkedlist_item_get_value(IGNORED_PTR_ARG))
+                .SetReturn(TEST_IOTHUB_SERVICE_FEEDBACK_RECORD_BUFFER + i * sizeof(IOTHUB_SERVICE_FEEDBACK_RECORD));
+            STRICT_EXPECTED_CALL(singlylinkedlist_get_next_item(IGNORED_PTR_ARG))
+                .SetReturn(i < (array_count - 1) ? TEST_LIST_ITEM_HANDLE : NULL);
+            STRICT_EXPECTED_CALL(gballoc_free(IGNORED_PTR_ARG));
+        }
         STRICT_EXPECTED_CALL(singlylinkedlist_destroy(IGNORED_PTR_ARG));
-
         STRICT_EXPECTED_CALL(gballoc_free(IGNORED_PTR_ARG));
-
         STRICT_EXPECTED_CALL(json_array_clear(IGNORED_NUM_ARG));
         STRICT_EXPECTED_CALL(json_value_free(IGNORED_NUM_ARG));
 
         //act
-        onMessageReceivedCallback(iothub_messaging_handle, TEST_MESSAGE_HANDLE);
+        void* amqp_result = (void*)saved_on_message_receiver_callback(iothub_messaging_handle, TEST_MESSAGE_HANDLE);
 
         //assert
-        ASSERT_ARE_EQUAL(int, IOTHUB_FEEDBACK_STATUS_CODE_SUCCESS, receivedFeedbackStatusCode);
         ASSERT_ARE_EQUAL(char_ptr, umock_c_get_expected_calls(), umock_c_get_actual_calls());
+        ASSERT_ARE_EQUAL(int, 1, saved_feedback_message_received_count);
+        IOTHUB_SERVICE_FEEDBACK_RECORD* feedbackRecord0 = ((IOTHUB_SERVICE_FEEDBACK_RECORD*)TEST_IOTHUB_SERVICE_FEEDBACK_RECORD_BUFFER);
+        IOTHUB_SERVICE_FEEDBACK_RECORD* feedbackRecord1 = ((IOTHUB_SERVICE_FEEDBACK_RECORD*)(TEST_IOTHUB_SERVICE_FEEDBACK_RECORD_BUFFER + sizeof(IOTHUB_SERVICE_FEEDBACK_RECORD)));
+        ASSERT_ARE_EQUAL(int, IOTHUB_FEEDBACK_STATUS_CODE_SUCCESS, feedbackRecord0->statusCode);
+        ASSERT_ARE_EQUAL(int, IOTHUB_FEEDBACK_STATUS_CODE_SUCCESS, feedbackRecord1->statusCode);
 
         ///cleanup
-        IoTHubMessaging_LL_Close(iothub_messaging_handle);
-        IoTHubMessaging_LL_Destroy(iothub_messaging_handle);
     }
 
     TEST_FUNCTION(IoTHubMessaging_LL_FeedbackMessageReceived_happy_path_feedback_expired)
     {
         //arrange
-        IOTHUB_MESSAGING_HANDLE iothub_messaging_handle = IoTHubMessaging_LL_Create(TEST_IOTHUB_SERVICE_CLIENT_AUTH_HANDLE);
-        (void)IoTHubMessaging_LL_Open(iothub_messaging_handle, TEST_FUNC_IOTHUB_OPEN_COMPLETE_CALLBACK, (void*)1);
-        (void)IoTHubMessaging_LL_SetFeedbackMessageCallback(iothub_messaging_handle, TEST_FUNC_IOTHUB_FEEDBACK_MESSAGE_RECEIVED_CALLBACK, (void*)1);
+        IOTHUB_MESSAGING_HANDLE iothub_messaging_handle = create_messaging_handle();
+        ASSERT_ARE_EQUAL(int, 0, open_messaging_handle(iothub_messaging_handle, false));
 
-        receivedFeedbackStatusCode = IOTHUB_FEEDBACK_STATUS_CODE_UNKNOWN;
+        int feedbackContext1 = 50000001;
```

Comment:
nit: magic number

Reply:
Not necessarily. It's just a random number in a test. But I'll create a define for that.

Reply:
Changed.